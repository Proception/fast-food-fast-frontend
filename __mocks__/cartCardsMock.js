import yam from '../assets/images/yam.jpg';
import rice from '../assets/images/rice.jpg';
import fufu from '../assets/images/fufu.jpg';
import { removeItemFromCart, updateCartTotal, updateItemQuantity, makeOrder } from '../src/actions/orderActions';

const props = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: (e) => {},
  onChange: () => {},
  order: {
    cart: [
      {
        "id": 1,
        "imageUrl": yam,
        "price": 500,
        "title": "Boiled Yam",
      },
      {
        "id": 2,
        "imageUrl": rice,
        "price": 900,
        "title": "Rice",
      },
      {
        "id": 3,
        "imageUrl": fufu,
        "price": 500,
        "title": "Fufu",
      },
    ],
    cartTotalPrice: 0
  },
  auth: {
    isLoading: true,
    user: {
      id: 1
    }
  },
  removeItemFromCart,
  updateCartTotal,
  updateItemQuantity,
  makeOrder
};

const emptyCartProps = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: (e) => {},
  onChange: () => {},
  order: {
    cart: [],
    cartTotalPrice: 0
  },
  auth: {
    isLoading: false,
    user: {
      id: 1
    }
  },
  removeItemFromCart,
  updateCartTotal,
  updateItemQuantity,
  makeOrder
};
export { props, emptyCartProps };
  