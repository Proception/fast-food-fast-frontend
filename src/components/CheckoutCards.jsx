import React, { Component } from 'react';

const CheckoutCards = ({ item, onClick, onChange }) => (
  <div className="selected-items" key={item.id}>
    <div className="item-details">
      <p>
        <label htmlFor="foodname">Food Name :<span>{item.title}</span></label>
      </p>
      <p>
        <label htmlFor="price">Price per unit :<span>â‚¦{item.price}</span></label>
      </p>
      <p>
        <label htmlFor="quantity">Qty :
                <input id={item.id} onChange={onChange} type="number" name="1" min="1" max="100" step="1" defaultValue={item.quantity}/></label>
      </p>
    </div>
    <div className="item-commands">
      <p>
        <img src={item.imageUrl} />
      </p>
      <p>
        <input onClick={onClick} className="button" type="button" value="Delete Item" name="" id={item.id} />
      </p>
    </div>
  </div>
);
export default CheckoutCards;
