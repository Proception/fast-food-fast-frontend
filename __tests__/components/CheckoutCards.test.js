import React from 'react';
import { shallow } from 'enzyme';
import CheckoutCards from '../../src/components/CheckoutCards.jsx';
import { props }  from '../../__mocks__/cartCardsMock.js';

let wrapper;
beforeEach(() => {
  const { item } = props;
  wrapper = shallow(<CheckoutCards item={item}/>);
});
describe('CheckoutCards Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
