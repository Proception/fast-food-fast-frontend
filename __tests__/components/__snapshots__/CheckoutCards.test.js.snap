// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutCards Component should match snapshot 1`] = `
<div
  id="sec"
>
  <div
    className="selected-items"
    key="1"
  >
    <div
      className="item-details"
    >
      <p>
        <label
          htmlFor="foodname"
        >
          Food Name :
          <span>
            Boiled Yam
          </span>
        </label>
      </p>
      <p>
        <label
          htmlFor="price"
        >
          Price per unit :
          <span>
            ₦
            500
          </span>
        </label>
      </p>
      <p>
        <label
          htmlFor="quantity"
        >
          Qty :
          <input
            max="100"
            min="0"
            name="1"
            step="1"
            type="number"
            value="1"
          />
        </label>
      </p>
    </div>
    <div
      className="item-commands"
    >
      <p>
        <img
          src="test-file-stub"
        />
      </p>
      <p>
        <input
          className="button"
          id={1}
          name=""
          type="button"
          value="Delete Item"
        />
      </p>
    </div>
  </div>
  <div
    className="selected-items"
    key="2"
  >
    <div
      className="item-details"
    >
      <p>
        <label
          htmlFor="foodname"
        >
          Food Name :
          <span>
            Rice
          </span>
        </label>
      </p>
      <p>
        <label
          htmlFor="price"
        >
          Price per unit :
          <span>
            ₦
            900
          </span>
        </label>
      </p>
      <p>
        <label
          htmlFor="quantity"
        >
          Qty :
          <input
            max="100"
            min="0"
            name="1"
            step="1"
            type="number"
            value="1"
          />
        </label>
      </p>
    </div>
    <div
      className="item-commands"
    >
      <p>
        <img
          src="test-file-stub"
        />
      </p>
      <p>
        <input
          className="button"
          id={2}
          name=""
          type="button"
          value="Delete Item"
        />
      </p>
    </div>
  </div>
  <div
    className="selected-items"
    key="3"
  >
    <div
      className="item-details"
    >
      <p>
        <label
          htmlFor="foodname"
        >
          Food Name :
          <span>
            Fufu
          </span>
        </label>
      </p>
      <p>
        <label
          htmlFor="price"
        >
          Price per unit :
          <span>
            ₦
            500
          </span>
        </label>
      </p>
      <p>
        <label
          htmlFor="quantity"
        >
          Qty :
          <input
            max="100"
            min="0"
            name="1"
            step="1"
            type="number"
            value="1"
          />
        </label>
      </p>
    </div>
    <div
      className="item-commands"
    >
      <p>
        <img
          src="test-file-stub"
        />
      </p>
      <p>
        <input
          className="button"
          id={3}
          name=""
          type="button"
          value="Delete Item"
        />
      </p>
    </div>
  </div>
</div>
`;
